apps/trading/client-pages/home/home.tsx

```diff
@@ -17,18 +21,37 @@ export const Home = () => {
   const { data, error, loading } = useDataProvider({
     dataProvider: marketsWithDataProvider,
   });
-  const { update } = useGlobalStore((store) => ({
+  const { update, marketId } = useGlobalStore((store) => ({
     update: store.update,
+    marketId: store.marketId,
   }));
-
+  const variables = useMemo(
+    () => ({
+      marketId: marketId || '',
+    }),
+    [marketId]
+  );
+  const { data: singleMarketData, loading: singleMarketLoading } =
```

Comment:
suggestion (non blocking): For simplicity I think it would be better to not use the data provider here and just let the page load with an invalid market, and have the user see the 'Not found' message. It'll be more predictable and more obvious to tell the user that their expected market is gone rather than redirect them somewhere else

Reply:
Hm, I didn't assume we want to redirect user to possibly not existing market. Yeah, I'm gonna rearrange this checking here